diff --git a/examples/c/mfcalc/mfcalc.y b/examples/c/mfcalc/mfcalc.y
index 4227d5c..13f93c0 100644
--- a/examples/c/mfcalc/mfcalc.y
+++ b/examples/c/mfcalc/mfcalc.y
@@ -21,6 +21,9 @@
 %{
   #include <stdio.h>  /* For printf, etc. */
   #include <math.h>   /* For pow, used in the grammar. */
+  #include <stdlib.h> /* malloc, realloc - must be before parser code. */
+  #include <string.h> /* strlen, strcmp, strdup. */
+  #include <assert.h>
   #include "calc.h"   /* Contains definition of 'symrec'. */
   int yylex (void);
   void yyerror (char const *);
@@ -108,9 +111,6 @@ init_table (void)
    always succeed, and that integer calculations
    never overflow.  Production-quality code should
    not make these assumptions.  */
-#include <assert.h>
-#include <stdlib.h> /* malloc, realloc. */
-#include <string.h> /* strlen. */
 
 symrec *
 putsym (char const *name, int sym_type)
