diff --git a/examples/c/rpcalc/rpcalc.y b/examples/c/rpcalc/rpcalc.y
index bc67c58..2efcac5 100644
--- a/examples/c/rpcalc/rpcalc.y
+++ b/examples/c/rpcalc/rpcalc.y
@@ -23,6 +23,7 @@
 %{
   #include <stdio.h>
   #include <math.h>
+  #include <stdlib.h>
   int yylex (void);
   void yyerror (char const *);
 %}
@@ -59,7 +60,6 @@ exp:
    and tabs, and returns 0 for end-of-input. */
 
 #include <ctype.h>
-#include <stdlib.h>
 
 int
 yylex (void)
