diff --git a/data/skeletons/yacc.c b/data/skeletons/yacc.c
index 64b9ac6..9a51b23 100644
--- a/data/skeletons/yacc.c
+++ b/data/skeletons/yacc.c
@@ -567,13 +567,13 @@ b4_push_if([], [b4_lac_if([], [[
 #  endif
 #  ifndef YYMALLOC
 #   define YYMALLOC malloc
-#   if ! defined malloc && ! defined EXIT_SUCCESS
+#   if ! defined malloc && ! defined EXIT_SUCCESS && ! defined __ZOSLIB_H
 void *malloc (YYSIZE_T); /* INFRINGES ON USER NAME SPACE */
 #   endif
 #  endif
 #  ifndef YYFREE
 #   define YYFREE free
-#   if ! defined free && ! defined EXIT_SUCCESS
+#   if ! defined free && ! defined EXIT_SUCCESS && ! defined __ZOSLIB_H
 void free (void *); /* INFRINGES ON USER NAME SPACE */
 #   endif
 #  endif
