diff --git a/lib/ffsl.h b/lib/ffsl.h
index 9774510..11d9b5a 100644
--- a/lib/ffsl.h
+++ b/lib/ffsl.h
@@ -50,6 +50,7 @@ ffs (int i)
 int
 FUNC (TYPE i)
 {
+#ifndef __MVS__
 #if __GNUC__ > 3 || (__GNUC__ == 3 && __GNUC_MINOR__ >= 4) \
     || (__clang_major__ >= 4)
   return GCC_BUILTIN (i);
@@ -62,6 +63,7 @@ FUNC (TYPE i)
   else
     return 0;
 #else
+#endif
   unsigned TYPE j = i;
   /* Split j into chunks, and look at one chunk after the other.  */
   enum { chunk_bits = CHAR_BIT * sizeof (unsigned int) };
@@ -88,5 +90,7 @@ FUNC (TYPE i)
   /* Last loop round.  k = chunk_count - 1.  */
   return (chunk_count - 1) * chunk_bits
          + ffs ((unsigned int) (j >> ((chunk_count - 1) * chunk_bits)));
+#ifndef __MVS__
+#endif
 #endif
 }
